src/main/java/org/zalando/nakadi/domain/SubscriptionCursor.java

```diff
@@ -40,8 +43,8 @@ public boolean equals(final Object o) {
     @Override
     public int hashCode() {
         int result = super.hashCode();
-        result = 31 * result + eventType.hashCode();
-        result = 31 * result + cursorToken.hashCode();
+        result = 31 * result + (eventType != null ? eventType.hashCode() : 0);
```

Comment:
This check for null in hashCode is a hack for validation to work. If I put that check to equals as well - then why shouldn't I put the null check to all other places?
That's why I don't want to put this check to any other places.